utils/release.ts

```diff
@@ -0,0 +1,61 @@
+import { HarmonyRelease, HarmonyTrack, ReleaseGroupType } from '@/harmonizer/types.ts';
+
+/** Guess the types for a release from release and track titles. */
+export function guessTypesForRelease(release: HarmonyRelease) {
+	let types = release.types || new Set();
+	types = types.union(guessTypesFromTitle(release.title));
+	if (!types.has('Live') && guessLiveRelease(release.media.flatMap((media) => media.tracklist))) {
+		types.add('Live');
+	}
+	release.types = types;
+}
+
+const detectTypesPatterns = [
+	// Commonly used for Bandcamp releases
+	/\s\((EP|Single|Live|Demo)\)(?:\s\(.*?\))?$/i,
+	// iTunes singles and EPs
+	/\s- (EP|Single|Live)(?:\s\(.*?\))?$/i,
+	// Generic "EP" suffix
+	/\s(EP)(?:\s\(.*?\))?$/i,
+];
+
+const typeMap: Map<string, ReleaseGroupType> = new Map([
+	['demo', 'Demo'],
+	['ep', 'EP'],
+	['live', 'Live'],
+	['single', 'Single'],
+]);
+
+/** Guesses a release type from a title. */
+export function guessTypesFromTitle(title: string): Set<ReleaseGroupType> {
+	const types = new Set<ReleaseGroupType>();
+	detectTypesPatterns.forEach((pattern) => {
+		const match = title.match(pattern);
+		if (match) {
+			const type = match[1].toLowerCase();
+			types.add(typeMap.get(type) as ReleaseGroupType);
+		}
+	});
+	return types;
+}
+
+/** Returns true if all track titles indicate a live release. */
+export function guessLiveRelease(tracks: HarmonyTrack[]): boolean {
+	return tracks?.length > 0 && tracks.every((track) => {
+		const types = guessTypesFromTitle(track.title);
+		return types.has('Live');
+	});
+}
```

Comment:
Per [the style guidelines](https://musicbrainz.org/doc/Style/Release_Group#Secondary_types) secondary types "should apply to most of the tracks on the releases within the release group. It's OK if some of the tracks do not fit the type (for example, if a live album has some bonus studio tracks), as long as the type applies to the releases overall."
That means we could even use a threshold of 50% (or more) of the tracks here.

Reply:
I thought about this. But this again goes too much into guesswork. [This release](https://musicbrainz.org/release/115cd763-3c7b-4e3c-83ff-004b2d1074cb) has 5 tracks, 4 of which are live. It is (rightly) not classified as a live release, but a single which happens to have 4 live tracks as bonus.

Reply:
That makes sense, I haven't thought about such edge cases.